### PR TITLE
Rubaiyat/integrate caseflow with virtual hearings

### LINKED ISSUE #12334: Integrate Virtual Hearing Modal with Controller
### Background
#12333 created a modal that appears after a user switches from central to virtual hearings and vice versa, we now need to send `virtual_hearings` params to the HearingsController (#12332) 

### Objective
There will be 2 requests when virtual hearings data changes. One to the virtual_hearing endpoint and another to the hearings endpoint. These requests should be chained in that order. If the virtual hearing request fails, the user is presented with a failure and no hearings data will be saved. If the hearings request fails, it's OK that the virtual hearing data was saved, we simply want the error message to accurately reflect what failed. 

- [ ] Set value of hearing type dropdown based on `hearing` data
- [ ] Add method that posts virtual_hearing params when a hearing is changed from anything to `virtual`

params:
```
{
  virtual_hearing: {
     hearing_id: 123,
     hearing_type: "",
     veteran_email: "",
     judge_email: "",
     representative_email: "",
     active: true
  }
}
```
- [ ] Add method that sends `{ virtual_hearing: { active: false } }` to endpoint when a hearing is changed from `virtual` to `central` or `video`

### PR DESCRIPTION
**Pull request by rubaiyat22**
Resolves #12334
Relates to #12519
### Description
- integrate changes on hearing details page with backend, sends patch request to hearings_controller
- integrate changes on daily docket row with backed, sends patch request to hearings_controller
- modify tests on hearing_prep_spec.rb and hearing_details_spec.rb

### CODE REVIEW

File: client/app/hearings/containers/DailyDocketContainer.jsx [hunk 1]
```diff
@@ -98,6 +98,18 @@ export class DailyDocketContainer extends React.Component {
     });
   };
 
+  formatVirtualHearing = (hearing) => {
```
Comment (rubaiyat22) [inline client/app/hearings/containers/DailyDocketContainer.jsx]: this function is not working as expected but left it as a placeholder for future changes

File: app/serializers/hearings/hearing_serializer.rb [hunk 2]
```diff
@@ -71,6 +71,11 @@ class HearingSerializer
   attribute :veteran_gender, if: for_full
   attribute :veteran_last_name
   attribute :virtual, &:virtual?
```
Comment (lomaxap) [inline app/serializers/hearings/hearing_serializer.rb]: i think we need to change this to is_virtual

File: client/app/hearings/components/dailyDocket/DailyDocketRow.jsx [hunk 3]
```diff
@@ -256,14 +267,15 @@ class HearingActions extends React.Component {
         {this.getRightColumn()}
       </div>
       {/* This is where the modal for the edit virtual hearing modal is supposed to appear*/}
-      {(user.userCanScheduleVirtualHearings && this.state.virtualHearingModalActive) &&
+      {(user.userCanScheduleVirtualHearings && this.state.virtualHearingModalActive && hearing.isVirtual) &&
```
Comment (lomaxap) [inline client/app/hearings/components/dailyDocket/DailyDocketRow.jsx]: i'm surprised this works. Does `hearing.isVirtual` return anything but `undefined`?

File: client/app/hearings/containers/DailyDocketContainer.jsx (hunk 1 above)
Comment (lomaxap) [reply client/app/hearings/containers/DailyDocketContainer.jsx]: will be covered with #12519

File: client/app/hearings/containers/DailyDocketContainer.jsx [hunk 4]
```diff
@@ -98,6 +98,18 @@ export class DailyDocketContainer extends React.Component {
...
+    if (!virtualHearing.isVirtual){
```
Comment (lomaxap) [inline client/app/hearings/containers/DailyDocketContainer.jsx]: this should be `hearing.isVirtual`

**Pull request merged**